activerecord/test/cases/relations_test.rb

```diff
@@ -1057,25 +1059,40 @@ def test_many
     posts = Post.all
 
     assert_queries(2) do
-      assert posts.many? # Uses COUNT()
+      assert_sql(/SELECT\ +1/) do
+        assert posts.many?
+      end
       assert posts.many? { |p| p.id > 0 }
       assert_not posts.many? { |p| p.id < 2 }
     end
 
     assert_predicate posts, :loaded?
+
+    assert_no_queries do
+      assert posts.many? # Uses loaded @records
+    end
+
+    posts = Post.where(author_id: 1)
+    assert posts.many?
+    assert_not_predicate posts, :loaded?
+
+    posts = Post.where(id: 1)
+    assert_not posts.many?
   end
 
   def test_many_with_limits
     posts = Post.all
 
     assert_predicate posts, :many?
-    assert_not_predicate posts.limit(1), :many?
+    assert_sql(/SELECT\ +.posts.\.\*/) do
```

Comment:
Same here, is there a reason why we can't do `/SELECT  "posts"\.\*/`?

Reply:
@gmcgibbon PostgreSQL and MySQL have different quotation. In MySQL it'd be SELECT \`posts\`.*